src/main/java/picard/sam/SamErrorMetric/ReadBaseStratification.java

```diff
@@ -1014,4 +1233,84 @@ private static int stratifyMappingQuality(final SAMRecord sam) {
     private static String stratifyReadGroup(final SAMRecord sam) {
         return sam.getReadGroup().getReadGroupId();
     }
-}
+
+    private static Integer stratifyIndelLength(final RecordAndOffset recordAndOffset, final SAMLocusAndReference locusInfo, CollectSamErrorMetrics.BaseOperation operation) {
+        if(operation != CollectSamErrorMetrics.BaseOperation.Insertion && operation != CollectSamErrorMetrics.BaseOperation.Deletion) {
+            return null;
+        }
+
+        CigarElement cigarElement = getIndelElement(recordAndOffset, operation);
+        if(cigarElement == null) {
+            // No CIGAR operation for given position.
+            return null;
+        }
+        if(operation == CollectSamErrorMetrics.BaseOperation.Insertion && cigarElement.getOperator() != CigarOperator.I) {
+            log.warn("Wrong CIGAR operator for the given position. This is an error and should be fixed.");
```

Comment:
This case should never occur (if it does, it would mean an error in the CIGAR string or an error in the code, i.e. an error by me), however, it is seems like a sensible check to make. I now throw an exception instead of a log warning, let me know if you have a better suggestion